pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -20,7 +20,10 @@
 # TODO enable shared libs for cross-compiling
 , enableSharedExecutables ? !isCross && (((ghc.isGhcjs or false) || stdenv.lib.versionOlder "7.7" ghc.version))
 , enableSharedLibraries ? !isCross && (((ghc.isGhcjs or false) || stdenv.lib.versionOlder "7.7" ghc.version))
-, enableSplitObjs ? !stdenv.isDarwin # http://hackage.haskell.org/trac/ghc/ticket/4013
+# For GHC 8.0.2 we use split-sections instead
+# Unsupported for Darwin/Clang, see http://hackage.haskell.org/trac/ghc/ticket/4013
+, enableSplitObjs ? (!stdenv.isDarwin && (stdenv.lib.versionAtLeast "8.0.1" ghc.version))
+, enableSplitSections ? (!stdenv.isDarwin && (stdenv.lib.versionOlder "8.0.1" ghc.version))
```

Comment:
We could enable all of these with a single argument and disable those situations that don't work. If somone wants a different strategy, they can just disable and pass some `--ghc-option` flags. Thoughts?